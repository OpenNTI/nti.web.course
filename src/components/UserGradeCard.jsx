import React from 'react';
import PropTypes from 'prop-types';
import {Avatar, DisplayName} from 'nti-web-commons';
import { scoped } from 'nti-lib-locale';

const t = scoped('course.components.UserGradeCard', {
	assignGrade: 'Assign Final Grade',
	notAvailable: '-'
});

export default class UserGradeCard extends React.Component {
	static propTypes = {
		user: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
		grade: PropTypes.number,
		onAssignGrade: PropTypes.func,
		assignable: PropTypes.bool
	}

	onAssignClick = () => {
		const { onAssignGrade } = this.props;

		onAssignGrade && onAssignGrade();
	}

	renderAssignGrade () {
		if(this.props.assignable) {
			return <div className="assign-grade" onClick={this.onAssignClick}>{t('assignGrade')}</div>;
		}

		return <div className="sub-info">{this.props.user}</div>;
	}

	render () {
		const { user, grade } = this.props;

		return (
			<div className="user-grade-card">
				<div className="avatar-container">
					<Avatar className="avatar" entity={user}/>
					<div className="grade-container">
						<div className="grade">{grade || t('notAvailable')}</div>
					</div>
				</div>
				<div className="user-name"><DisplayName entity={user}/></div>
				{this.renderAssignGrade()}
			</div>
		);
	}
}
