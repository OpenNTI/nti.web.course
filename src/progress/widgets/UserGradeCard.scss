@import "~@nti/style-common";

.user-grade-card {
	width: 252px;
	height: 251px;
	background-color: #fbfbfb;
	text-align: center;

	.avatar-container {
		position: relative;

		.avatar {
			width: 100px;
			height: 100px;
			border-radius: 50px;
			margin-top: 40px;
		}

		.grade-container {
			color: #3fb3f6;
			position: absolute;
			right: 78px;
			bottom: 6px;
			border: solid 2px #fbfbfb;
			border-radius: 20px;
			width: 34px;
			font-size: 14px;
			height: 34px;
			background-color: #fbfbfb;
			padding-top: 2px;
			padding-left: 2px;

			.grade {
				border: solid 2px #3fb3f6;
				border-radius: 25px;
				width: 27px;
				height: 27px;
				padding-top: 2px;
			}
		}
	}

	.user-name {
		color: var(--primary-grey);
		font-size: 17px;
		text-overflow: ellipsis;
		overflow: hidden;
		margin: 19px 19px 5px;
	}

	.sub-info {
		font-size: 14px;
		color: #b8b8b8;
		font-weight: 500;
	}

	.assign-grade {
		color: $primary-blue;
		font-size: 14px;
		cursor: pointer;
	}
}
