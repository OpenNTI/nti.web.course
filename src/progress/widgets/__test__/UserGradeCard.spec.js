import { render, act } from '@testing-library/react';

import * as TestUtils from '@nti/web-client/test-utils';
import { flushPromises } from '@nti/lib-commons/test-utils';

import UserGradeCard from '../UserGradeCard';

const { tearDownTestClient, setupTestClient } = TestUtils;

const getMockService = () => {
	return {};
};

const onBefore = () => {
	jest.useFakeTimers();
	setupTestClient(getMockService());
};

const onAfter = () => {
	tearDownTestClient();
};

/* eslint-env jest */
describe('Progress widgets user grade card test', () => {
	beforeEach(() => onBefore());
	afterEach(onAfter);

	test('With grade', async () => {
		const cmp = render(<UserGradeCard user="testuser" grade={76} />);

		await act(async () => {
			jest.runAllTimers();
			await flushPromises();
			jest.runAllTimers();
		});

		expect(cmp.asFragment()).toMatchSnapshot();
	});

	test('Without grade', async () => {
		const cmp = render(<UserGradeCard user="testuser" />);

		await act(async () => {
			jest.runAllTimers();
			await flushPromises();
			jest.runAllTimers();
		});

		expect(cmp.asFragment()).toMatchSnapshot();
	});
});
