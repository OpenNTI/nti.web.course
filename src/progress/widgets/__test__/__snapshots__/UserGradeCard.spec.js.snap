// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress widgets user grade card test With grade 1`] = `
<DocumentFragment>
  <div
    class="user-grade-card"
  >
    <div
      class="avatar-container"
    >
      <img
        alt="Avatar for Anonymous User"
        class="avatar avatar avatar-color-6 avatar"
        src="unresolved-user-avatar.png"
      />
      <div
        class="grade-container"
      >
        <div
          class="grade"
        >
          76
        </div>
      </div>
    </div>
    <div
      class="user-name"
    >
      <span
        class="username"
        rel="author"
      >
        Anonymous User
      </span>
    </div>
    <div
      class="sub-info"
    >
      testuser
    </div>
  </div>
</DocumentFragment>
`;

exports[`Progress widgets user grade card test Without grade 1`] = `
<DocumentFragment>
  <div
    class="user-grade-card"
  >
    <div
      class="avatar-container"
    >
      <img
        alt="Avatar for Anonymous User"
        class="avatar avatar avatar-color-6 avatar"
        src="unresolved-user-avatar.png"
      />
      <div
        class="grade-container"
      >
        <div
          class="grade"
        >
          -
        </div>
      </div>
    </div>
    <div
      class="user-name"
    >
      <span
        class="username"
        rel="author"
      >
        Anonymous User
      </span>
    </div>
    <div
      class="sub-info"
    >
      testuser
    </div>
  </div>
</DocumentFragment>
`;
